### ISSUE TITLE
Receive urlopen error on Windows platform

### ISSUE THREAD

**Issue by adelosa** [Yelp/swagger_spec_validator#99]
Works as expected on versions prior to and including 2.1.0. Broken on 2.2.0 and 2.3.0.
Receive the following URLerror exception when validating file with absolute path on Windows system.
<urlopen error unknown url type: c>
"c" being the c:\ in the filename

```
  File "C:\Users\adelo\venv\lib\site-packages\swagger_spec_validator\validator20.py", line 120, in validate_json
    schema = read_file(schema_path)
  File "C:\Users\adelo\venv\lib\site-packages\swagger_spec_validator\common.py", line 39, in read_file
    return read_url(urljoin('file://', file_path))
  File "C:\Users\adelo\venv\lib\site-packages\swagger_spec_validator\common.py", line 43, in read_url
    with contextlib.closing(urlopen(url, timeout=timeout)) as fh:
  File "C:\Python27\Lib\urllib2.py", line 154, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Python27\Lib\urllib2.py", line 429, in open
    response = self._open(req, data)
  File "C:\Python27\Lib\urllib2.py", line 452, in _open
    'unknown_open', req)
  File "C:\Python27\Lib\urllib2.py", line 407, in _call_chain
    result = func(*args)
  File "C:\Python27\Lib\urllib2.py", line 1266, in unknown_open
    raise URLError('unknown url type: %s' % type)
SwaggerValidationError: ('<urlopen error unknown url type: c>', URLError(u'unknown url type: c',))
```

**Comment by alexandruionica** [comment]
I have encountered the same issue.  Windows 2016 64bit using Python 3.6 is throwing the above while Linux doesn't.
`Traceback (most recent call last):
  File "C:\Users\OBFUSCATED_PATH\.venv\lib\site-packages\swagger_spec_validator\common.py", line 22, in wrapper
    return method(*args, **kwargs)
  File "C:\Users\OBFUSCATED_PATH\.venv\lib\site-packages\swagger_spec_validator\validator20.py", line 121, in 
validate_json
    schema = read_file(schema_path)
  File "C:\Users\OBFUSCATED_PATH\.venv\lib\site-packages\swagger_spec_validator\common.py", line 39, in read_file
    return read_url(urljoin('file://', file_path))
  File "C:\Users\OBFUSCATED_PATH\.venv\lib\site-packages\swagger_spec_validator\common.py", line 43, in read_url
    with contextlib.closing(urlopen(url, timeout=timeout)) as fh:
  File "c:\python36\Lib\urllib\request.py", line 223, in urlopen
    return opener.open(url, data, timeout)
  File "c:\python36\Lib\urllib\request.py", line 526, in open
    response = self._open(req, data)
  File "c:\python36\Lib\urllib\request.py", line 549, in _open
    'unknown_open', req)
  File "c:\python36\Lib\urllib\request.py", line 504, in _call_chain
    result = func(*args)
  File "c:\python36\Lib\urllib\request.py", line 1388, in unknown_open
    raise URLError('unknown url type: %s' % type)
urllib.error.URLError: <urlopen error unknown url type: c>`

**Comment by macisamuele** [comment]
Thanks a lot for the issue reporting, the fix has been released in `swagger-spec-validator==2.3.1`
